### PR TITLE
ARM32: fix switch jump table generation bug

### PR DESCRIPTION
**Pull request by BredPet**
- see https://github.com/dotnet/coreclr/issues/13194

Signed-off-by: Petr Bred <bredpetr@gmail.com>

### CODE REVIEW

**Comment by BredPet** [comment]
@Dmitri-Botcharnikov 
@sergign60 
@jkotas please review

**Comment by jkotas** [comment]
cc @dotnet/jit-contrib 

I am not sure whether `PLATFORM_UNIX` is the preferred ifdef for this kind of change in the RyuJIT codebase.

LGTM otherwise.

File: src/jit/codegenlegacy.cpp [hunk 1]
```diff
@@ -15387,7 +15387,14 @@ void CodeGen::genTableSwitch(regNumber reg, unsigned jumpCnt, BasicBlock** jumpT
...
-    if (jumpCnt < minSwitchTabJumpCnt)
+    if (jumpCnt < minSwitchTabJumpCnt ||
+#ifdef PLATFORM_UNIX
+        // Force using an inlined jumping instead switch table generation
+        // See https://github.com/dotnet/coreclr/issues/13194
+        (compiler->IsTargetAbi(CORINFO_CORERT_ABI)))
+#else
+        0)
+#endif
     {
```
Comment (BruceForstall) [inline src/jit/codegenlegacy.cpp]: Use `_TARGET_UNIX_`, not `PLATFORM_UNIX`. Also, the bug says this issue is only for ARM. Also, please don't simply reference the bug: write the full description here, as a comment, for why this case exists. Thus, I would expect to see:

```
    bool useJumpSequence = jumpCnt < minSwitchTabJumpCnt;

#if defined(_TARGET_UNIX_) && defined(_TARGET_ARM_)
    // Force using inlined jumping instead switch table generation.
    // << insert full description of WHY here >>     
    // See also [https://github.com/dotnet/coreclr/issues/13194](https://github.com/dotnet/coreclr/issues/13194).
    useJumpSequence = useJumpSequence || compiler->IsTargetAbi(CORINFO_CORERT_ABI);
#endif // defined(_TARGET_UNIX_) && defined(_TARGET_ARM_)

    if (useJumpSequence)
    ...
```

**Comment by BredPet** [comment]
@jkotas 
@BruceForstall 
please review again

**Pull request merged**